.travis.yml

```diff
@@ -1,11 +1,21 @@
 language: r
 sudo: required
 cache: packages
+dist: trusty
 
 apt_packages:
-  - r-cran-rgdal
   - libproj-dev
 
+before_install:
+  - sudo add-apt-repository ppa:ubuntugis/ubuntugis-unstable --yes
+  - sudo apt-get --yes --force-yes update -qq
+  - sudo apt-get install --yes libudunits2-dev libproj-dev libgeos-dev libgdal-dev
+
+r_packages:
```

Comment:
I think sf (or some of its depends/imports, e.g. udunits2) specifically needs the dev version of these system libraries to be available and I don't think travis installs those by default.